client/my-sites/people/invite-people/index.jsx

```diff
@@ -135,41 +135,34 @@ class InvitePeople extends React.Component {
 		createInviteValidation( this.props.siteId, filteredTokens, role );
 
 		if ( filteredTokens.length > usernamesOrEmails.length ) {
-			analytics.tracks.recordEvent( 'calypso_invite_people_token_added' );
+			this.props.recordTracksEvent( 'calypso_invite_people_token_added' );
 		} else {
-			analytics.tracks.recordEvent( 'calypso_invite_people_token_removed' );
+			this.props.recordTracksEvent( 'calypso_invite_people_token_removed' );
 		}
 	};
 
-	onMessageChange = event => {
-		this.setState( { message: event.target.value } );
-	};
+	onMessageChange = event => this.setState( { message: event.target.value } );
 
 	onRoleChange = event => {
 		const role = event.target.value;
 		this.setState( { role } );
 		createInviteValidation( this.props.siteId, this.state.usernamesOrEmails, role );
 	};
 
-	onFocusTokenField = () => {
-		analytics.tracks.recordEvent( 'calypso_invite_people_token_field_focus' );
-	};
+	onFocusTokenField = () =>
+		this.props.recordTracksEvent( 'calypso_invite_people_token_field_focus' );
 
-	onFocusRoleSelect = () => {
-		analytics.tracks.recordEvent( 'calypso_invite_people_role_select_focus' );
-	};
+	onFocusRoleSelect = () =>
+		this.props.recordTracksEvent( 'calypso_invite_people_role_select_focus' );
 
-	onFocusCustomMessage = () => {
-		analytics.tracks.recordEvent( 'calypso_invite_people_custom_message_focus' );
-	};
+	onFocusCustomMessage = () =>
+		this.props.recordTracksEvent( 'calypso_invite_people_custom_message_focus' );
```

Comment:
these are clear examples where calling `this.props.onFocusCustomMessage()` would be preferable to `this.onFocusCustomMessage` which then calls `recordTracksEvent`

Ideally we'd never have any analytics/metrics specifically called out inside the components. they are side-effects that can and will change over time while the semantics of the components will stay the same. put another way, they are triggered by activity which ideally all has a funnel through the dispatcher, so passing in a semantically-meaningful prop like `selectRole()` and then dispatching or wrapping a dispatch with analytics preserves the nice boundary we have established with our components.

Reply:
> these are clear examples where calling this.props.onFocusCustomMessage() would be preferable to this.onFocusCustomMessage which then calls recordTracksEvent

Ah, I should have retained that lesson from elsewhere ;)

Agree. I'll update accordingly.